### PR TITLE
Story 1 - List all books

### PR DESCRIPTION
**Pull request by GDTimi**
Submitting Story 1 for review - thanks!

- Get all books currently in the system
- Must match the format shown in the [documentation](https://github.com/iO-Academy/book-swap-api-docs/blob/main/README.md)

### CODE REVIEW

File: app/Http/Controllers/BookController.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?php
+
```
Comment (mporam) [inline app/Http/Controllers/BookController.php:2]: remove 1 empty line

File: app/Http/Controllers/BookController.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+class BookController extends Controller
+{
+
+    private Book $book;
+
+    public function __construct(Book $book) 
+    {
+        $this->book=$book;
+    }
+
```
Comment (mporam) [inline app/Http/Controllers/BookController.php:18]: remove 1 empty line

File: app/Http/Controllers/BookController.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    public function getAllBooks()
+    {
+        $books=$this->book->all();
```
Comment (mporam) [inline app/Http/Controllers/BookController.php:22]: missing spaces either side of the `=`

File: app/Http/Controllers/BookController.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+        foreach($books as $book) {
+            $book->genre->name;
+            $book->reviews;
+        }
+
+        return response()->json([
+            'message'=> 'Books successfully retrived', 
+            'data'=> $books
```
Comment (mporam) [inline app/Http/Controllers/BookController.php:31]: should have a space before the `=>`

**Comment by Kel-E-creator** [comment]
Hi Mike, 
no http rtests and your modelfatories are empty?
I had a query on the route tests as we have set up our database with one to many relationships.  Genre and Review are separate tables.  In the example tests, you mentioned there was a different way to test relationships, for this exercise do we just comment out the links to the Genre and Reviews in the product controller as we did in the example, then uncomment them to use the function?

File: app/Http/Controllers/BookController.php (hunk 1 above)
Comment (Sipho-Ndlovu) [reply app/Http/Controllers/BookController.php:2]: Lines have removed!

File: app/Http/Controllers/BookController.php (hunk 3 above)
Comment (Sipho-Ndlovu) [reply app/Http/Controllers/BookController.php]: Line has been removed!

**Pull request merged**